flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/AbstractCatalogView.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import org.apache.flink.table.api.TableSchema;
+
+import java.util.Map;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * An abstract catalog view.
+ */
+public abstract class AbstractCatalogView implements CatalogView {
+	// Original text of the view definition.
+	private final String originalQuery;
+
+	// Expanded text of the original view definition
+	// This is needed because the context such as current DB is
+	// lost after the session, in which view is defined, is gone.
+	// Expanded query text takes care of the this, as an example.
+	private final String expandedQuery;
+
+	private final TableSchema schema;
+	private final Map<String, String> properties;
+	private final String comment;
+
+	public AbstractCatalogView(String originalQuery, String expandedQuery, TableSchema schema,
+			Map<String, String> properties, String comment) {
+		this.originalQuery = checkNotNull(originalQuery, "originalQuery cannot be null");
```

Comment:
should also check whether the string is empty. I'll change for you when merging code